corporate/models.py

```diff
@@ -30,9 +36,27 @@ class Customer(models.Model):
     # they purchased.
     exempt_from_from_license_number_check: bool = models.BooleanField(default=False)
 
+    @property
+    def is_self_hosted(self) -> bool:
+        return self.remote_server is not None and self.realm is None
+
+    @property
+    def is_cloud(self) -> bool:
+        return self.realm is not None and self.remote_server is None
+
     def __str__(self) -> str:
         return f"<Customer {self.realm} {self.stripe_customer_id}>"
 
+    def clean(self) -> None:
+        if self.realm is None and self.remote_server is None:
+            raise ValidationError(_("Customer must link to an organization or a remote server."))
+        elif self.realm is not None and self.remote_server is not None:
+            raise ValidationError(
+                _(
+                    "Customer must have either link to an organization or a remote server, but not both."
+                )
+            )
```

Comment:
I don't think this `clean` method will do anything, since I suspect nothing will call it when creating objects.  I think we can skip having this method; structurally we'll ensure the model we want by only having methods for creating Customer objects that set exactly one of those properties.